src/platform_impl/macos/mod.rs

```diff
@@ -217,6 +220,33 @@ impl TrayIcon {
         }
         self.attrs.menu_on_left_click = enable;
     }
+
+    pub fn rect(&self) -> Option<Rect> {
+        let ns_status_item = self.ns_status_item?;
+        unsafe {
+            let button = ns_status_item.button();
+            let window: id = button.window();
+            if window.is_null() {
+                None
+            } else {
+                let rect_in_window: NSRect =
+                    msg_send![button, convertRect:button.bounds() toView:nil];
+                let screen_rect = window.convertRectToScreen_(rect_in_window);
+
+                Some(Rect {
+                    position: crate::dpi::PhysicalPosition::new(
+                        screen_rect.origin.x,
+                        screen_rect.origin.y,
+                    ),
+                    size: crate::dpi::PhysicalSize::new(
+                        screen_rect.size.width,
+                        screen_rect.size.height,
+                    )
```

Comment:
there is an implementation to get the rectangle on Line 357

Reply:
Picked it up. I imagine we could make a helper there to reuse the implementation.